core/src/main/java/com/sequenceiq/cloudbreak/service/LoadBalancerConfigService.java

```diff
@@ -157,6 +176,29 @@ public String getLoadBalancerUserFacingFQDN(Long stackId) {
         return loadBalancers;
     }
 
+    private boolean isNetworkUsingPrivateSubnet(Network network, EnvironmentNetworkResponse envNetwork) {
+        return isSelectedSubnetAvailableAndRequestedType(network, envNetwork, true);
+    }
+
+    private boolean isNetworkUsingPublicSubnet(Network network, EnvironmentNetworkResponse envNetwork) {
+        return isSelectedSubnetAvailableAndRequestedType(network, envNetwork, false);
+    }
+
+    private boolean isSelectedSubnetAvailableAndRequestedType(Network network, EnvironmentNetworkResponse envNetwork, boolean isPrivate) {
+        if (network != null) {
+            Json attributes = network.getAttributes();
+            Map<String, Object> params = attributes == null ? Collections.emptyMap() : attributes.getMap();
+            String subnetId = String.valueOf(params.get(SUBNET_ID));
```

Comment:
this might cause an NPE if params map does not have SUBNET_ID

Reply:
It won't cause an NPE. It will however convert a null value to the string "null", which isn't what I want to happen. I've changed this line to do a null check and added new unit tests to test that everything works when `network`, `network.getAttributes()`, and `params.get(SUBNET_ID)` are each null.

Reply:
no, actually String.valueOf(null) is indeed NPE

Reply:
The code of `String.valueOf` is:
```
    public static String valueOf(Object obj) {
        return obj == null ? "null" : obj.toString();
    }
```
That doesn't look like it'd throw an NPE to me?